Assets/_System/Scripts/TurnBasedScripts/CombatManager.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CombatManager : MonoBehaviour
+{
+    [Header("Selected Unit")]
+    public GridObject selectedUnit = null;   // Currently highlighted GridTile
```

Comment:
Like above, unless this is variable is meant to be modified by outside scripts, make sure to set it to private and to add a getter.

Reply:
Keep that public because it get's modified frequently and by other scripts